packages/intersection-observer/README.md

```diff
@@ -137,6 +137,9 @@ import { createVisibilityObserver } from "@solid-primitives/intersection-observe
 let el: HTMLDivElement | undefined;
 
 const useVisibilityObserver = createVisibilityObserver({ threshold: 0.8 });
+
+// ! extremely important to use an accessor for the element to not lose reactivity
+// ! typescript will give you a warning if you don't use an accessor
```

Comment:
1. there is no reactivity to be lost if `ref` is just a `let` variable. But it is important to wrap it in a thunk so that it could be evaluated lazily, inside createEffect, when the `ref` is already populated.
For example here you don't have to use the callback form:
```tsx
let el: HTMLDivElement | undefined;

const useVisibilityObserver = createVisibilityObserver({ threshold: 0.8 });

onMount(() => {
  const visible = useVisibilityObserver(el)
  
  createEffect(() => {
    console.log(visible())
  })
})
```
2. typescript won't warn you about it - I think. because the expected param type is `MaybeAccessor<Element | FalsyValue>`

Reply:
The following code did not work without an accessor function `() => element`:

```ts
<For each={Object.keys(messages())}>
  {(id) => {
    let element: HTMLDivElement | undefined;
    const isVisible = useVisibilityObserver(() => element);

    return (
      <div ref={element}>
        <Show when={isVisible()}>
          <Messages messages={messages()[id]} />
        </Show>
      </div>
    );
  }}
</For>;

```


https://user-images.githubusercontent.com/35429197/215616376-610e1966-7174-46db-a486-41c932df8dcc.mp4

Reply:
Your example uses useVisbilityObserver requires the `onMount` effect and, most likely, won't work otherwise. In short, reactivity is lost if you don't use an accessor function as element is and stays undefined.

Reply:
Regardin 2: Yes. TypeScript won't warn because the type is `Falsy` which is true since a ref is undefined on rendering. I couldn't come up with a better type. Thus, I added my note to the docs

Reply:
Looked at the type again. 

```ts
declare function createVisibilityObserver(options?: IntersectionObserverInit & {
    initialValue?: boolean;
}, setter?: MaybeAccessor<VisibilitySetter>)

type MaybeAccessor<T> = T | Accessor<T>
```

`MaybeAccessor` should probably be replaced with `Accessor` to avoid the unforeseen "bug" I encountered.

Reply:
> Your example requires the `onMount` effect and, most likely, won't work otherwise. In short, reactivity is lost if you don't use an accessor function as an element is and stays undefined if no external reactive mechanism sets `useVisibilityObserver` like `onMount`.

you are right, but I just wanted to be more specific about whats going on. It's not as simple as "reactivity is lost" beasue it has to nothing to do with reactivity - only timing. So the commnet is fair, I just would prefer it to explain more about whats going on, rather then simply say "don't do this".

> `MaybeAccessor` should probably be replaced with `Accessor` to avoid the unforeseen "bug" I **encountered.**

As I showed, using the primitive without function param is fine in cases where the element is present immediately. So I cannot just change the type to `Accessor`.
BUT it might be a good idea to print a warning to the console is the passed `target` is `undefined` and not an `accessor`. And the type could be changed to this, to be more helpful:
```ts
Accessor<Element | FalsyValue> | Element
```
with this it won't be possible to pass in variable that could be `undefined` without wrapping it in a thunk.